lib/electric_slide/call_queue.rb

```diff
@@ -62,7 +62,7 @@ def available_agent_summary
     # @return {Agent}
     def checkout_agent
       agent = @strategy.checkout_agent
-      agent.presence = :busy
+      agent.presence = :busy if agent
```

Comment:
I know you made other changes, but simply checking for a nil agent here on this line still means that this method can return a nil agent, which should not be possible.  But I also don't think we should be adding the protections here at all. If a nil agent is in the queue, then something upstream has failed, and other badness is likely to happen.